framework/generate_doc.mjs

```diff
@@ -0,0 +1,116 @@
+import * as fs from 'fs';
+import data from './.jsii.tabl.json' assert { type: 'json' };
+
+// select only snippets for readme files
+const snippets = Object.fromEntries(
+    Object.entries(data.snippets)
+        .filter(([_,v]) => v.location.api.api === 'moduleReadme')
+);
+
+const linksMapping = {
+    '../storage/README.md#datalakestorage' : '../02-Storage/03-data-lake-storage.mdx',
+    '#datacatalogdatabase': './data-catalog-database'
+}
+
+let previousSubmodule = '';
+let previousStart = 0;
+let codeLength = 0;
+let fullReadme = {};
+
+let currentMDFile = '';
+let lines = [];
+
+for (let key in snippets) {
+    const snippet = snippets[key];
+    const typescript = snippet.translations.$.source;
+    const python = snippet.translations.python.source;
+
+    const submodule = snippet.location.api.moduleFqn.replace('aws-dsf.', '');
+    if (previousSubmodule !== submodule) {
+        previousSubmodule = submodule;
+        previousStart = 0;
+        codeLength = 0;
+    }
+    const file = `src/${submodule}/README.md`;
+
+    // do not read file on each snippet
+    if (file !== currentMDFile) {
+        currentMDFile = file;
+        const buffer= fs.readFileSync(file);
+        const str= buffer.toString();
+        lines = str.split("\n");
+    }
+
+    let line = snippet.location.field.line;
+    let realLine = line - codeLength;
+
+    // jsii does not count comments, we must add lines for each one
+    // let numberOfComments = lines.slice(0, realLine).filter(elt => elt.startsWith("[//]: # ")).length;
+    // realLine += numberOfComments;
+
+    // console.log(line + ' - ' + realLine);
+
+    if (!fullReadme[submodule]) {
+        fullReadme[submodule] = '';
+    }
+    fullReadme[submodule] += lines.slice(previousStart, realLine - 1).join('\n');
+    fullReadme[submodule] += `
+<Tabs>
+  <TabItem value="typescript" label="TypeScript" default>
+
+  \`\`\`typescript
+  ${typescript}
+  \`\`\`
+  
+  \`\`\`mdx-code-block
+  
+  </TabItem>
+  <TabItem value="python" label="Python">
+
+  \`\`\`python
+  ${python}
+  \`\`\`
+
+  </TabItem>
+</Tabs>\n
+`;
+
+    previousStart = realLine + 1;
+    codeLength += typescript.split("\n").length + 1;
+}
+
+const titleRegEx = new RegExp('\\[\\/\\/\]: # \\((.*)\\)\\n', 'gm');
+
+for (let module in fullReadme) {
+    const moduleReadme = fullReadme[module];
+    const constructReadmes = moduleReadme.split(titleRegEx);
+    let filename = '';
+    for (let i = 0; i < constructReadmes.length; i++) {
+        let constructReadme = constructReadmes[i];
+        if (constructReadme.startsWith(module)) {
+            filename = '../website/docs/constructs/library/generated/_' + constructReadme.replace('.', '-') + '.mdx';
```

Comment:
Hard-coded paths... Maybe parameterize the script or at least define as `const` in the beginning of the file?